### PR TITLE
[xray-exporter] Add adding of test cases to test execution

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-to-xray-exporter/src/main/java/org/vividus/xray/exporter/XrayExporter.java [hunk 1]
```diff
@@ -64,28 +64,29 @@
...
 
             for (Scenario scenario : story.getScenarios())
             {
-                exportScenario(story.getPath(), scenario);
+                exportScenario(story.getPath(), scenario).ifPresent(testCaseIds::add);
             }
         }
 
-        publishErrors();
+        afterExport(testCaseIds);
```
Comment (valfirst) [inline vividus-to-xray-exporter/src/main/java/org/vividus/xray/exporter/XrayExporter.java:78]:
```java
linkTestCasesToTestExecution(testCaseIds);
publishErrors();
```

File: vividus-to-xray-exporter/src/main/java/org/vividus/xray/facade/XrayFacade.java [hunk 2]
```diff
@@ -75,6 +79,17 @@ public void updateTestCase(String testCaseKey, TestCaseParameters testCaseParame
...
     }
 
+    public void updateTestExecution(String testExecutionKey, List<String> testCaseKeys) throws IOException
+    {
+        TestExecution testExecution = new TestExecution(testCaseKeys);
+        String testExecutionRequest = objectMapper.writeValueAsString(testExecution);
+        LOGGER.atInfo()
+              .addArgument(() -> StringUtils.join(testCaseKeys, ", "))
+              .addArgument(testExecutionKey)
+              .log("Add {} test cases to {} test execution");
```
Comment (valfirst) [inline vividus-to-xray-exporter/src/main/java/org/vividus/xray/facade/XrayFacade.java:89]: `Link` or `Add` ?

File: vividus-to-xray-exporter/src/main/java/org/vividus/xray/model/TestExecution.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    private final List<String> add;
+
+    public TestExecution(List<String> add)
+    {
+        this.add = add;
+    }
+
+    public List<String> getAdd()
+    {
+        return List.copyOf(add);
```
Comment (valfirst) [inline vividus-to-xray-exporter/src/main/java/org/vividus/xray/model/TestExecution.java:32]: just return it as is

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-to-xray-exporter/src/main/java/org/vividus/xray/facade/XrayFacade.java (hunk 2 above)
Comment (uarlouski) [reply vividus-to-xray-exporter/src/main/java/org/vividus/xray/facade/XrayFacade.java:89]: from API point of view add, I'll ask end users

Comment (uarlouski) [reply vividus-to-xray-exporter/src/main/java/org/vividus/xray/facade/XrayFacade.java:89]: decided to use Add

File: vividus-to-xray-exporter/src/main/java/org/vividus/xray/exporter/XrayExporter.java [hunk 4]
```diff
@@ -124,6 +128,15 @@ private TestCaseParameters createTestCaseParameters(String storyTitle, Scenario
         return parameters;
     }
 
+    private void linkTestCasesToTestExecution(List<String> testCaseIds) throws IOException
```
Comment (valfirst) [inline vividus-to-xray-exporter/src/main/java/org/vividus/xray/exporter/XrayExporter.java:131]: `linkTestCasesToTestExecution` -> `addTestCasesToTestExecution`

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**